### PR TITLE
Bootstrap and jquery added using webpacker

### PR DESCRIPTION
**Pull request by ArifaAbid**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,6 +8,8 @@ gem 'rails', '~> 6.0.3', '>= 6.0.3.2'
 # Use mysql2 as the database for Active Record
 gem 'mysql2', '>= 0.4.4'
 # Use Puma as the app server
+
+gem 'autoprefixer-rails'
```
Comment (bk-az) [inline Gemfile]: why we need this?

File: app/assets/stylesheets/application.css [hunk 2]
```diff
@@ -1,15 +1,2 @@
-/*
- * This is a manifest file that'll be compiled into application.css, which will include all the files
```
Comment (bk-az) [inline app/assets/stylesheets/application.css:2]: why removed this?

Comment (bk-az) [reply app/assets/stylesheets/application.css:2]: what is the purpose of 
*= require_tree .
 *= require_self

File: app/assets/stylesheets/application.css [hunk 3]
```diff
@@ -1,15 +1,2 @@
-/*
```
Comment (bk-az) [inline app/assets/stylesheets/application.css:1]: rename this file to scss, we wil use sass

File: app/assets/stylesheets/main.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+h1 {
```
Comment (bk-az) [inline app/assets/stylesheets/main.css]: remove changes done for testing purpose

File: app/controllers/home_controller.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class HomeController < ApplicationController
+
+
```
Comment (bk-az) [inline app/controllers/home_controller.rb]: index action

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
+  get 'try/index'
```
Comment (bk-az) [inline config/routes.rb]: why added these routes?
get 'try/index'
get 'new/about'

File: app/javascript/packs/application.js [hunk 7]
```diff
@@ -15,3 +17,8 @@ require("channels")
 //
 // const images = require.context('../images', true)
 // const imagePath = (name) => images(name, true)
+import 'bootstrap'
```
Comment (bk-az) [inline app/javascript/packs/application.js]: don't import css in js, use asset pipeline for css

File: app/javascript/packs/application.js [hunk 8]
```diff
@@ -2,11 +2,13 @@
 // present in this directory. You're encouraged to place your actual application logic in
 // a relevant structure within app/javascript and only use these pack files to reference
 // that code so it'll be compiled.
-
```
Comment (bk-az) [inline app/javascript/packs/application.js:5]: avoid stray changes

File: Gemfile [hunk 9]
```diff
@@ -27,7 +29,6 @@ gem 'jbuilder', '~> 2.7'
 
 # Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.4.2', require: false
-
```
Comment (bk-az) [inline Gemfile:30]: avoid stray changes

File: Gemfile [hunk 10]
```diff
@@ -8,6 +8,8 @@ gem 'rails', '~> 6.0.3', '>= 6.0.3.2'
 # Use mysql2 as the database for Active Record
 gem 'mysql2', '>= 0.4.4'
 # Use Puma as the app server
+
```
Comment (bk-az) [inline Gemfile]: in future if you need to add a gem, add a comment above it, currently you have added  autoprefixer gem below web-server comment

File: app/views/home/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,4 @@
+<h1>Hello World</h1>
+<button type="button" class="btn btn-secondary" data-toggle="tooltip" data-placement="top" title="Tooltip on top">
```
Comment (bk-az) [inline app/views/home/index.html.erb]: add meaningful content, like Employee Tracking System, otherwise remove everything that is added for testing

File: app/javascript/packs/application.js [hunk 12]
```diff
@@ -2,11 +2,13 @@
...
 require("@rails/ujs").start()
 require("turbolinks").start()
 require("@rails/activestorage").start()
 require("channels")
+require("packs/custom")
```
Comment (bk-az) [inline app/javascript/packs/application.js]: does this file exists?

File: app/javascript/packs/application.js [hunk 13]
```diff
@@ -2,11 +2,13 @@
...
+require("packs/posts.js")
+require("jquery")
```
Comment (bk-az) [inline app/javascript/packs/application.js]: why?

File: app/javascript/packs/application.js (hunk 12 above)
Comment (bk-az) [reply app/javascript/packs/application.js]: require("packs/custom")
require("packs/posts.js")

File: app/javascript/packs/application.js [hunk 14]
```diff
@@ -15,3 +17,8 @@ require("channels")
 //
 // const images = require.context('../images', true)
 // const imagePath = (name) => images(name, true)
+import 'bootstrap'
+
+document.addEventListener("turbolinks:load", () => {
```
Comment (bk-az) [inline app/javascript/packs/application.js]: require('bootstrap')

File: Gemfile [hunk 15]
```diff
@@ -27,7 +27,6 @@ gem 'jbuilder', '~> 2.7'
 
 # Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.4.2', require: false
-
```
Comment (bk-az) [inline Gemfile]: add line

File: app/assets/stylesheets/application.scss [hunk 16]
```diff
@@ -0,0 +1,3 @@
+@import '../../../node_modules/bootstrap/scss/bootstrap';
```
Comment (bk-az) [inline app/assets/stylesheets/application.scss]: @import '~bootstrap/scss/bootstrap';

File: app/assets/stylesheets/header.css [hunk 17]
```diff
@@ -0,0 +1,27 @@
+ .navbar {
```
Comment (bk-az) [inline app/assets/stylesheets/header.css]: remove this file, use bootstrap

File: app/assets/stylesheets/home.css [hunk 18]
```diff
@@ -0,0 +1,8 @@
+h1 {
```
Comment (bk-az) [inline app/assets/stylesheets/home.css]: remove this file use bootstrap

File: app/controllers/application_controller.rb [hunk 19]
```diff
@@ -1,2 +1,5 @@
 class ApplicationController < ActionController::Base
+  def index
```
Comment (bk-az) [inline app/controllers/application_controller.rb]: why?

File: app/javascript/packs/application.js (hunk 12 above)
Comment (bk-az) [reply app/javascript/packs/application.js]: not resolved

File: app/assets/stylesheets/application.scss (hunk 16 above)
Comment (ArifaAbid) [reply app/assets/stylesheets/application.scss]: This line ~bootstrap/scss/bootstrap not working and giving error.May b here is path issue.But @import '../../../node_modules/bootstrap/scss/bootstrap'; working fine.

File: config/webpack/environment.js [hunk 20]
```diff
@@ -1,3 +1,11 @@
 const { environment } = require('@rails/webpacker')
 
+const webpack = require('webpack')
+environment.plugins.append('Provide',
+  new webpack.ProvidePlugin({
+    $: 'jquery',
```
Comment (bk-az) [inline config/webpack/environment.js]: what is the purpose of this change? if you need to require poper.js and jquery

File: app/javascript/packs/application.js (hunk 13 above)
Comment (ArifaAbid) [reply app/javascript/packs/application.js]: While adding popper.js and jquery with bootstrap I think so that may be thses both files are must need to require while writting require('bootstrap').But now I have understand that all three are independent.

Comment (bk-az) [reply app/javascript/packs/application.js]: no, bootstrap depends on poper.js and jquery.js

File: config/webpack/environment.js (hunk 20 above)
Comment (ArifaAbid) [reply config/webpack/environment.js]: The reason is same I was following bootstrap using webpacker link in which this changes was required.Thats why I was confused while require('bootstrap') and think all these are required for bootstrap.But now I have understand.

File: app/views/home/index.html.erb [hunk 21]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
```
Comment (bk-az) [inline app/views/home/index.html.erb]: why?

File: app/views/home/index.html.erb [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
```
Comment (bk-az) [inline app/views/home/index.html.erb]: why doing this?

File: app/views/home/index.html.erb [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
```
Comment (bk-az) [inline app/views/home/index.html.erb]: ??

File: app/views/home/index.html.erb [hunk 24]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html>
+<head>
```
Comment (bk-az) [inline app/views/home/index.html.erb]: why we need to add head tag here? what is the purpose of layout file,

https://github.com/zhamza372/EmployeeManagementSystem/blob/0380598e8f0d09c8d6f99b2c2831295ebce29805/app/views/layouts/application.html.erb

File: app/views/home/index.html.erb (hunk 21 above)
Comment (ArifaAbid) [reply app/views/home/index.html.erb]: I have removed home.css so added this stylesheet for styling purposes.Currently working on styling.

Comment (bk-az) [reply app/views/home/index.html.erb]: why importing bootstrap in application.js and application.scss then?

File: app/views/home/index.html.erb (hunk 23 above)
Comment (ArifaAbid) [reply app/views/home/index.html.erb]: min.js is minified version of complete bootstrap style(CSS) and and component(JS).May help me while designing UI.

Comment (bk-az) [reply app/views/home/index.html.erb]: then what is the purpose of require("bootstrap") in application.js

Comment (ArifaAbid) [reply app/views/home/index.html.erb]: understood now.

File: app/views/home/index.html.erb (hunk 21 above)
Comment (ArifaAbid) [reply app/views/home/index.html.erb]: understood now.

File: app/views/home/index.html.erb (hunk 22 above)
Comment (ArifaAbid) [reply app/views/home/index.html.erb]: understood now.

File: app/views/home/_header.html.erb [hunk 25]
```diff
@@ -0,0 +1,35 @@
+<nav class="navbar navbar-expand-xl navbar-light bg-warning">
```
Comment (bk-az) [inline app/views/home/_header.html.erb]: `<nav class="navbar navbar-expand-lg navbar-dark bg-dark">`

File: app/views/home/index.html.erb [hunk 26]
```diff
@@ -0,0 +1 @@
+<h2 class="text-danger text-center">Employee Managment System</h2>
```
Comment (bk-az) [inline app/views/home/index.html.erb]: `text-dark`
keep it dark for now

File: app/views/home/_header.html.erb (hunk 25 above)
Comment (ArifaAbid) [reply app/views/home/_header.html.erb]: No need to change color???I was just want to look it beautiful.

File: app/views/home/index.html.erb (hunk 26 above)
Comment (ArifaAbid) [reply app/views/home/index.html.erb]: Ok
